tests/phpunit/tests/rest-api/rest-attachments-controller.php

```diff
@@ -1923,6 +1923,25 @@ public function test_rest_insert_attachment_hooks_fire_once_on_create() {
 		$this->assertSame( 1, self::$rest_after_insert_attachment_count );
 	}
 
+	/**
+	 * Verify that the naming behavior of REST media uploads matches core media uplaods.
+	 * In particular filenames with spaces should maintain the spaces rather than being replaced by hyphens.
+	 *
+	 * @ticket 57957
+	 */
+	public function test_rest_upload_filename_spaces() {
+		wp_set_current_user( self::$editor_id );
+		$request = new WP_REST_Request( 'POST', '/wp/v2/media' );
+		$request->set_header( 'Content-Type', 'image/jpeg' );
+		$request->set_header( 'Content-Disposition', 'attachment; filename=Filename With Spaces.jpg' );
+		$request->set_body( file_get_contents( self::$test_file ) );
+
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+		$this->assertSame( 201, $response->get_status() );
+		$this->assertSame( 'Filename With Spaces', $data['title']['raw'] );
```

Comment:
```suggestion
		$this->assertSame( 201, $response->get_status(), 'The file was not uploaded.' );
		$this->assertSame( 'Filename With Spaces', $data['title']['raw'], 'An incorrect filename was returned.' );
```
Test methods with multiple assertions should have a final `$message` parameter to provide context when an assertion fails. [Reference](https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#using-assertions).

Even though existing test methods in this file lack the `$message` parameter, adding it in this test method reduces our work later and sets a precedent for other new tests to do the same to meet standards.